crates/rustfix/src/replace.rs

```diff
@@ -297,28 +303,35 @@ mod tests {
 
     #[test]
     fn broken_replacements() {
-        let mut d = Data::new(b"foo");
+        let mut d = Data::new("foo");
         assert!(matches!(
-            d.replace_range(4..8, b"lol").unwrap_err(),
+            d.replace_range(4..8, "lol").unwrap_err(),
             Error::DataLengthExceeded(std::ops::Range { start: 4, end: 8 }, 3),
         ));
     }
 
     #[test]
-    fn replace_same_twice() {
-        let mut d = Data::new(b"foo");
-        d.replace_range(0..1, b"b").unwrap();
-        d.replace_range(0..1, b"b").unwrap_err();
-        assert_eq!("boo", str(&d.to_vec()));
+    fn insert_same_twice() {
+        let mut d = Data::new("foo");
+        d.replace_range(1..1, "b").unwrap();
+        assert_eq!("fboo", str(&d.to_vec()));
+        assert!(matches!(
+            d.replace_range(1..1, "b").unwrap_err(),
+            Error::AlreadyReplaced {
+                is_identical: true,
+                ..
+            },
+        ));
+        assert_eq!("fboo", str(&d.to_vec()));
```

Comment:
This test change should move to the second commit maybe.

Reply:
While this has already covered the case of <https://github.com/rust-lang/cargo/issues/13027>, I feel we should generate a snapshot test from the exact macro diagnostics, like <https://github.com/rust-lang/cargo/blob/de5832fe8fe1a05e278eee82bbc801d544ed4bd1/crates/rustfix/tests/everything/E0178.rs>.

I can help with that if it is too much a burden.

Reply:
I think I can handle it. I wanted to ask, though, do you know a good way to capture the original issue I raised, since it's `clippy` producing those suggestions?

Reply:
See <https://github.com/rust-lang/cargo/blob/7421ccf9159d5ac43883c53abe41ef1da657acfd/crates/rustfix/tests/parse_and_replace.rs?plain=1#L1-L21>.

Though I believe it is already fixed via <https://github.com/rust-lang/cargo/pull/13728>>. Granted your solution is better 👍🏾 

Just opened https://github.com/rust-lang/cargo/pull/14765 to switch and close it, so you don't need to worry setting up a test.